### PR TITLE
WIP: JSONAPI prefix

### PR DESCRIPTION
**Pull request by n2ygk**
## Description of the Change
Per a [discussion](https://github.com/django-json-api/django-rest-framework-json-api/pull/455#pullrequestreview-144719067) with @sliverc, renamed recently-added paginations class from `JsonApi`-prefix to `JSONAPI`-prefix. This includes new deprecation warnings for the old names.

I also added a minor fix to tox.ini not to run flake8 on the entire directory but only the relevant files.
You can cherry-pick that out if you prefer it not to be part of this PR.

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [x] unit-test added (no change; just class renamed)
- [x] documentation updated
- [x] changelog entry added to `CHANGELOG.md`
- [x] author name in `AUTHORS`

### CODE REVIEW

**Comment by n2ygk** [comment]
Dammit every time I try a rebase I get something backward and now it shows 12 files being changed.

**Comment by n2ygk** [comment]
```In general the way to get the best of both worlds is to rebase local changes you’ve made but haven’t shared yet before you push them in order to clean up your story, but never rebase anything you’ve pushed somewhere.``` -- https://git-scm.com/book/en/v2/Git-Branching-Rebasing

OK, merge it is from now on.

File: tox.ini [hunk 1]
```diff
@@ -20,7 +20,7 @@ commands =
 
 [testenv:flake8]
 deps = flake8
-commands = flake8
+commands = flake8 --exclude=migrations rest_framework_json_api example
```
Comment (sliverc) [inline tox.ini]: flake8 configuration is to be found in the `setup.cfg`. I think for consistency we should move all options there.

Comment (n2ygk) [reply tox.ini]: I'll clean up my mess. Just found another small error in the deprecation warnings as well.

**Comment by n2ygk** [comment]
closing this PR so I can undo the rebase message and resubmit.

**Pull request closed without merging**